Makefile.uk

```diff
@@ -61,22 +67,32 @@ LIBPYTHON3_SRC = $(LIBPYTHON3_ORIGIN)/$(LIBPYTHON3_BASENAME)
 # Library includes
 ################################################################################
 LIBPYTHON3_GLOBAL_INCLUDES-y += -I$(LIBPYTHON3_BASE)/include
-LIBPYTHON3_GLOBAL_INCLUDES-y += -I$(LIBPYTHON3_SRC)
 LIBPYTHON3_GLOBAL_INCLUDES-y += -I$(LIBPYTHON3_SRC)/Include
-CINCLUDES-y   += $(LIBPYTHON3_GLOBAL_INCLUDES-y)
-CXXINCLUDES-y += $(LIBPYTHON3_GLOBAL_INCLUDES-y)
+
+LIBPYTHON3_COMMON_INCLUDES-y += -I$(LIBPYTHON3_SRC)
+LIBPYTHON3_COMMON_INCLUDES-y += -I$(LIBPYTHON3_SRC)/Include/internal
+
+CINCLUDES-$(CONFIG_LIBPYTHON3) += $(LIBPYTHON3_GLOBAL_INCLUDES-y)
+CXXINCLUDES-$(CONFIG_LIBPYTHON3) += $(LIBPYTHON3_GLOBAL_INCLUDES-y)
 
 ################################################################################
 # Common flags
 ################################################################################
 # Suppress flags
-LIBPYTHON3_SUPPRESS_FLAGS += -Wno-unused-parameter \
-	-Wno-unused-variable -Wno-unused-value -Wno-unused-function \
-	-Wno-missing-field-initializers -Wno-implicit-fallthrough \
-	-Wno-cast-function-type -Wno-stringop-truncation \
-	-Wno-char-subscripts -Wno-sign-compare -Wno-maybe-uninitialized
-LIBPYTHON3_SUPRESS_CFLAGS-y   += $(LIBPYTHON3_SUPPRESS_FLAGS) -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
-LIBPYTHON3_SUPRESS_CXXFLAGS-y += $(LIBPYTHON3_SUPPRESS_FLAGS)
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-unused-parameter
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-unused-variable
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-unused-value
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-unused-function
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-missing-field-initializers
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-implicit-fallthrough
+LIBPYTHON3_SUPPRESS_FLAGS-$(call clang_version_ge,13,0) += -Wno-cast-function-type
+LIBPYTHON3_SUPPRESS_FLAGS-$(call have_gcc) += -Wno-stringop-truncation
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-char-subscripts
+LIBPYTHON3_SUPPRESS_FLAGS-y += -Wno-sign-compare
+LIBPYTHON3_SUPPRESS_FLAGS-$(call have_gcc) += -Wno-maybe-uninitialized
+LIBPYTHON3_SUPPRESS_FLAGS-$(call have_clang) += -Wno-documentation-unknown-command
+LIBPYTHON3_SUPRESS_CFLAGS-y   += $(LIBPYTHON3_SUPPRESS_FLAGS-y) -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
+LIBPYTHON3_SUPRESS_CXXFLAGS-y += $(LIBPYTHON3_SUPPRESS_FLAGS-y)
```

Comment:
I'd avoid more than one flag per line unless all flags act as an inseparable whole, otherwise it will make future diffs more unwieldy.
Adding large amounts of whitespace only to align long lines on the assignment operator also looks ugly IMO, so I think cleanest would be to group the conditional flags at the end.